lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -638,21 +638,16 @@ protected void search(List<LeafReaderContext> leaves, Weight weight, Collector c
     // threaded...?  the Collector could be sync'd?
     // always use single thread:
     for (LeafReaderContext ctx : leaves) { // search each subreader
-      final LeafCollector leafCollector;
-      try {
-        leafCollector = collector.getLeafCollector(ctx);
-      } catch (CollectionTerminatedException e) {
-        // there is no doc of interest in this reader context
-        // continue with the following leaf
-        continue;
-      }
       BulkScorer scorer = weight.bulkScorer(ctx);
```

Comment:
Should we also update javadocs for `Collector.getLeafCollector` that it will (may?) not be called when query can determine there will be no hits in this segment?